beep/generate_protocol.py

```diff
@@ -988,9 +989,76 @@ def convert_velocity_to_power_waveform(waveform_file, velocity_units):
 
     power = -power * df.v  # positive power = charge
 
-    return pd.DataFrame({'time(s)': df.t,
-                         'power_unscaled(W)': power,
-                         'power_scaled': power / max(abs(power))})
+    return pd.DataFrame({'time': df.t,
+                         'power': power})
+
+
+def generate_maccor_waveform_file(df, file_prefix, file_directory, mwf_config=None):
+    """
+    Helper function that takes in a variable power waveform and outputs a maccor waveform file (.MWF), which is read by
+    the cycler when the procedure file has a "fast-wave" step.
+    Relevant parameters to generate the .mwf files governing the input mode, charge/discharge limits, end conditions
+    and scaling are stored in /conversion_schemas/maccor_waveform_conversion.yaml
+
+    Args:
+        df (pd.DataFrame): power waveform containing two columns "time" and "power", in sec and W respectively.
+        file_prefix (str): prefix for the filename (extension is .MWF by default)
+        file_directory (str): folder to store the mwf file
+        mwf_config (dict): dictionary of instrument control parameters for generating the waveform
+
+    Returns:
+         path to the maccor waveform file generated
+    """
+
+    if mwf_config is None:
+        mwf_config = MACCOR_WAVEFORM_CONFIG
+
+    df['power'] = df['power'] / max(abs(df['power'])) * mwf_config['value_scale']
+
+    df['step_counter'] = df['power'].diff().fillna(0).ne(0).cumsum()
```

Comment:
Is this incrementing every time the power changes or flips sign?

Reply:
Increments every time the power changes.